packages/renderer/src/lib/statusbar/ProviderWidget.svelte

```diff
@@ -14,31 +16,63 @@ interface Props {
 
 let { entry, command = (): void => router.goto('/preferences/resources') }: Props = $props();
 
-let tooltipText = $derived.by(() => {
-  let tooltip = '';
+let connectionsStatuses = $derived.by(() => {
+  let statuses: ProviderStatus | ProviderConnectionStatus[];
+  let connectionsStatuses: { status: ProviderStatus | ProviderConnectionStatus; connecions: string }[] = [];
   if (entry.containerConnections.length > 0) {
-    tooltip = entry.containerConnections.map(c => c.name).join(', ');
+    statuses = [...new Set(entry.containerConnections.map(connection => connection.status))];
+    let connections = '';
+    for (const status of statuses) {
```

Comment:
Set is iterable [Set#Symbol.iterator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/Symbol.iterator)

So you can do the following

````ts
const statuses: Set<string> = new Set(entry.containerConnections.map(connection => connection.status)))
for(const status of statuses) {
	....
}
````

Reply:
![image](https://github.com/user-attachments/assets/5d7a64ae-fbf2-4315-bbda-a90417bc4429)

I was curious, and I used `https://jsbench.me/` to compare if spreading the set was faster/slower than using the build in iterator, result: we should use the internal iterator!